### ISSUE TITLE
Support for spring-data-redis 1.x

### ISSUE THREAD

**Issue by Hippoom** [TheBund1st/daming#18]
Some of the method usage of `RestTemplate` only available since `spring-data-redis` 2.x. This makes it difficult for projects still using `spring-data-redis` 1.x to adopt `daming`

**Comment by Hippoom** [comment]
Switching `org.springframework.data:spring-data-redis` from `2.1.X` to `1.8.X` causes error 

```
No such method: .set(K k, V v, Duration timeUnit)
```
Because the method is introduced since `2.1`. This could be fixed by using `.set(K k, V v, long timeout, TimeUnit timeUnit)`  which exists in `1.8.x`

**Comment by Hippoom** [comment]
If the RedisAdapters is written with `RestTemplate` class in `1.8.X`  
and `org.springframework.data:spring-data-redis` is introduced as `runtime` in the project, it causes this following error:

```
java.lang.NoSuchMethodError: org.springframework.data.redis.core.RedisTemplate.delete(Ljava/lang/Object;)V
	at com.thebund1st.daming.redis.RedisSmsVerificationRepository.remove(RedisSmsVerificationRepository.java:64)
```
Because the method without returning is removed in `2.X` and therefore makes it not feasible to keep compatibility.

Consider introduce an interface to decouple the delete operation with specific version of `RestTemplate`

**Comment by Hippoom** [comment]
See this [wiki](https://github.com/TheBund1st/daming/wiki/Working-with-Spring-Boot-1.x)

**Comment by Hippoom** [comment]
Released in `0.9.5`
